design/open/lakefs-kv-execution-plan.md

```diff
@@ -0,0 +1,72 @@
+# lakeFS on KV - Execution Plan
+
+Key/value data storage package (kv) that will supply key/value access to data to replace the current storage and locking done by Postgres.
+High level overview of the interface described in https://github.com/treeverse/lakeFS/blob/ed55edc2c24b24ed4e6fcccad1ae95844d38b9ad/design/open/metadata_kv/index.md. The package will support multiple back-end implementations.
+
+## Transform from using db to kv package
+
+High level interface described in the kv proposal, we should consider taking the DynamoDB interface to address the way we communicate `set-if` functionality while marshal the same value without fetching data first.
+
+Key/value format can be done using ProtoBuf / JSON.
+
+- key format based on path - a format with escaping should be supported to enable easy way to encode values as part of the key
+- each model should include version as part of the model information
+- migrate from old model to new one without downtime
```

Comment:
I think we need to consider the migration as a 2 step scenario and understand how / if support both steps without downtime.
1. Migration from db to kv on postgres
2. Migration from kv on postgres to another db
relevant for both OSS and cloud I think?

Reply:
Migration is from table format to kv format. The first step is to migrate all the sql based implementation to kv.
The second part is to support to instruct how we migrate changes when all the data is found in kv.
Moving from one underlying implementation to another (ex: postgres to dynamodb) will be done by dump and restore.